code/modules/shuttle/on_move.dm

```diff
@@ -401,13 +401,14 @@ All ShuttleMove procs go here
 
 	//Unload the planet
 	var/obj/docking_port/stationary/S = old_dock
-	for(var/datum/star_system/SYS in SSstarmap.star_systems)
-		var/datum/planet/PL = SYS.get_planet_for_z(oldT.z)
-		if(S.name == "FOB Landing Zone" && SSstarmap.current_planet.name != PL.name)
+	if(S.name != "FOB Landing Zone")
+		return
+	for(var/SYS in SSstarmap.star_systems)
```

Comment:
I still really hate this loop through all star_systems. Store the star system in the fob dock instead.

Reply:
Also, the previous review of this code:

~~Check name first before looping through every system in the game.~~

~~Make it a typeless loop.~~

You don't need to check where the planet is. If the no unload reason was set by you on landing you must remove it when you leave.

Preferably put this in beforeShuttleMove